### ISSUE TITLE
Can't filter out a category using the filter

### ISSUE THREAD

**Issue by Gonzom** [Js41637/Overwatch-Item-Tracker#95]
I was trying to use the filter to filter-out the achievement/special items from the all items list, but found out that no matter what combination i used, they still show up.

Maybe change the logic of how the filter works?
At start everything is marked, and you filter out those categories you don't want?

**Comment by Js41637** [comment]
I don't think this would be ideal but if you only wanted to view event items for a hero (or all-class) you'd have to deselect everything. Only option would be to have an exclude filter but i cant think of a nice way to add one.

**Comment by corwinburdick** [comment]
What about having the filter checkbox have three states:

Blank: Current default behavior. Doesn't affect what shows up.
Checked/Plus: Hides all items that don't match that filter
Crossed/Minus: Hides all items that match that filter

Clicking the checkbox could cycle through the three states in that order

**Comment by Js41637** [comment]
Yeah, maybe. Not as easy cause a checkbox only has 2 states, there is no inbetween in Javascript. Probably will require a bit of work to make something like this happen.

**Comment by ThePsionic** [comment]
You can probably kinda sorta hack around that with [indeterminate checkboxes](https://css-tricks.com/indeterminate-checkboxes/)

**Comment by Js41637** [comment]
They are visual only

**Comment by ThePsionic** [comment]
Fair enough.

**Comment by Gonzom** [comment]
I've thought about this a bit more and think I understand the concept problem here.

**Background**
We have 2 types of items - 
1) Real groups - these are items from an event ("Uprising") or PvP items/
2) Custom groups - these are groups we created out of the former group - "Pachi", "Achievement", "Hero Icons".

While an item can never be from both group 1, it can belong to group 1 and 2.

**Use cases**
Ok, so now how a user might want to view items.

First option would be to say "Show me all the items I'm missing ("Unselected") from "Uprising" and "Summer Games" - this will work fine and currently works.

Second option would be to say "Show me all items I'm missing ("Unselected") from "Uprising" and "Summer Games" which are of "Hero Icons" - this will work fine and currently works.

Third option would be to say "Show me all items I'm missing ("Unselected") from "Uprising" and "Summer Games" which are of "Hero Icons" and "Pachi" - this SHOULD work, but it doesn't as your response from #118  says it uses AND, which can never be.

**Suggestion**
1) Divide the filter into 3 sections instead of the current 4:
1a) "Selected"/"Unselected"
1b) Custom groups - "Achievement/Special", "Hero Icons", "Pachi"
1c) Event groups, "Competitive" and "Regular" (which should be for all non-event non-pvp items) - place "Regular" first, "Comp" second and the events after.

2) Change logic for sections b and c to "OR" instead of "AND".

Note: By adding the "Regular" group, you will make filtering out events possible in another way, as now someone can select "Regular" and just add to it the other filters.

Note 2: If this is fixed, issue #118 should be closed also.

**Comment by Gonzom** [comment]
Once issue #221 is completed this could now be done with a small tweak to it.

1. Divide the filter into 3 sections:
1a) "Selected"/"Unselected"
1b) Real groups (by order): "Regular Items", "Special", "Competitive", "Overwatch League", "Summer Games", "Halloween Terror", "Lunar New Year", "Overwatch Archives", "Anniversary"
1c) Custom Groups (by order): "Achievement", "Hero Icons", "Pachi", "Diablo", "Hearthstone", "Heroes of the Storm, "Starcraft", "World of Warcraft"

2. Change logic for sections b and c to "OR" instead of "AND".

3. Make selections in group "1b" work like "1a" as in a toggle option - just one could be selected. Entries from Group "1b" are unique in that they are always one or another.
